openapi/nexus.json

```diff
@@ -7831,18 +7831,18 @@
             "nullable": true,
             "type": "string"
           },
-          "make_primary": {
-            "description": "Make a secondary interface the instance's primary interface.\n\nIf applied to a secondary interface, that interface will become the primary on the next reboot of the instance. Note that this may have implications for routing between instances, as the new primary interface will be on a distinct subnet from the previous primary interface.\n\nNote that this can only be used to select a new primary interface for an instance. Requests to change the primary interface into a secondary will return an error.",
-            "default": false,
-            "type": "boolean"
-          },
           "name": {
             "nullable": true,
             "allOf": [
               {
                 "$ref": "#/components/schemas/Name"
               }
             ]
+          },
+          "primary": {
+            "description": "Make a secondary interface the instance's primary interface.\n\nIf applied to a secondary interface, that interface will become the primary on the next reboot of the instance. Note that this may have implications for routing between instances, as the new primary interface will be on a distinct subnet from the previous primary interface.\n\nNote that this can only be used to select a new primary interface for an instance. Requests to change the primary interface into a secondary will return an error.",
+            "default": false,
```

Comment:
@bnaecker is is true that the `default` is false? E.g. submitting without a value will try to make it _not_ the primary interface? Just wondering if a PUT was issued for the primary interface to rename it if the default would cause the request to fail by trying to make it _not_ the primary.

Reply:
It's true that the default is `false`. The value is ignored if the target interface is the primary. I.e., whether you've got true or false there, and that was specified by the client or left as the default, it's never touched if we're updating the primary interface.

Reply:
I realize now I never updated that docstring. It's not true that it returns an error, it's just ignored. It's not the best behavior, but generating an error condition was trickier than I thought in the database query. Would you mind changing that from `will return an error` to `will be ignored`?